tests/components/caldav/test_calendar.py

```diff
@@ -1140,3 +1161,92 @@ async def test_config_entry_supported_components(
     # No entity created when no components exist
     state = hass.states.get("calendar.calendar_4")
     assert not state
+
+
+@pytest.mark.parametrize(
+    ("service_data", "expected_ics_fields"),
+    [
+        # Basic event with all fields
+        (
+            {
+                "summary": "Test Event",
+                "start_date_time": datetime.datetime(2025, 8, 6, 10, 0, 0),
+                "end_date_time": datetime.datetime(2025, 8, 6, 11, 0, 0),
+                "description": "Test Description",
+                "location": "Test Location",
+            },
+            {
+                "description": "Test Description",
+                "dtend": datetime.datetime(2025, 8, 6, 11, 0),
+                "dtstart": datetime.datetime(2025, 8, 6, 10, 0),
+                "location": "Test Location",
+                "summary": "Test Event",
+            },
+        ),
+        # Event with only required fields
+        (
+            {
+                "summary": "Required Only",
+                "start_date_time": datetime.datetime(2025, 8, 7, 9, 0, 0),
+                "end_date_time": datetime.datetime(2025, 8, 7, 10, 0, 0),
+            },
+            {
+                "summary": "Required Only",
+                "dtstart": datetime.datetime(2025, 8, 7, 9, 0),
+                "dtend": datetime.datetime(2025, 8, 7, 10, 0),
+            },
+        ),
+        # All-day event (date only)
+        (
+            {
+                "summary": "All Day Event",
+                "start_date": datetime.date(2025, 8, 8),
+                "end_date": datetime.date(2025, 8, 9),
+            },
+            {
+                "summary": "All Day Event",
+                "dtstart": datetime.date(2025, 8, 8),
+                "dtend": datetime.date(2025, 8, 9),
+            },
+        ),
+        # Rrule is not supported in API (async_call) calls.
+    ],
+)
+@pytest.mark.parametrize("tz", [UTC])
+async def test_add_vevent(
+    hass: HomeAssistant,
+    setup_platform_cb: Callable[[], Awaitable[None]],
+    calendars: list[Mock],
+    service_data: dict,
+    expected_ics_fields: list[str],
+) -> None:
+    """Test adding a VEVENT to the calendar."""
+    await setup_platform_cb()
+
+    calendars[0].add_event = MagicMock(return_value=[])
+    await hass.services.async_call(
+        "calendar",
+        "create_event",
+        service_data,
+        target={"entity_id": TEST_ENTITY},
+        blocking=True,
+    )
+    await hass.async_block_till_done()
+
+    calendars[0].add_event.assert_called_once()
+    assert calendars[0].add_event.call_args
+    resulting_ics = calendars[0].add_event.call_args[1]
+
+    missing_fields = [
```

Comment:
What is the intent behind these tests? They look pretty odd and not very useful since its not checking anything about the values added. Please see the todo test `test_add_item`.

Reply:
i... look at it 5 times, and it looks like I wasn't checking things I was intending... I will fix it

My intention was for tests to check:
1. Check if all fields that are expected are in the ICS
2. Check if field values in ICS are those that we expect 
3. Check if uid is set and it's a valid UUID

Reply:
Just make it work like todo which is much simpler? I've also added a comment where the request parameters are created that I think will have a side effect of simplifying the test by only setting present fields.